### DISCUSSION TITLE
use the regex match of subdomain

### DISCUSSION THREAD

**Discussion by FrancYescO** [umputun/reproxy#169] [Q&A]
i'm trying witouth success to proxy a range of subdomain to another host/path.. something like this 

`reproxy --static.enabled --static.rule="(.*).example.com,(.*),https://$1.foo.com/$2"`

that should result in 
`http://AAAA.example.com/UUUU` proxied to `https://AAAA.foo.com/UUUU`

Is this possible somehow?

**Comment by umputun** [comment]
moved to discussion

**Comment by umputun** [comment]
You expect it to pass the matched group from the server and propagate to the destination, and this is not how it works because 
the regex match is for source->destination only; it doesn't work for the case you have (server->destination). See https://github.com/umputun/reproxy/blob/fa23778d4266d1e661e3cb33b546f53806f5d5a8/app/discovery/discovery.go#L181-L181

The regex in the server name is to match multiple servers to the same rule. However, it has nothing to do with source-to-destination matching at all.

_Making a change to support it should be relatively easy, but I don't know if this is a good idea because it looks very confusing. It treats the server and source as a single regex for the destination but simultaneously as two distinct matching groups for the host and source. In addition to this confusion, we will need a way to enforce the current behavior (regex server match and substitution from src to dst) and ensure it doesn't break back compatibility._

**Comment by FrancYescO** [comment]
_with a static provider: reproxy --static.enabled --static.rule="example.com/api/(.*),https://api.example.com/$1"_

This was taken from the docs and is missing the host parameter, idk is it is a typo... maybe is the best solution also for backward compatibility: if host is missing (2 parameter instead of 3 splitted by the comma) use the 1st parameter to match also the host (and get the ability to use both group match in the destination)

**Reply by umputun** [reply]
yeah, I missed the first parameter in docs, fixed

>  if host is missing (2 parameter instead of 3 splitted by the comma) use the 1st parameter to match also the host (and get the ability to use both group match in the destination)

interesting idea. in this case, we will need to infer the host from the source parameter and also make sure this host part is included in the match. 

If you want to make a PR I'll accept it in case it doesn't add much complexity (I don't think it will) and won't break back compatibility

**Reply by FrancYescO** [reply]
unfortunately i don't have a ready to go dev enviroment that allow me fast testing.. i'll go for a PR if i figure out how to setup it..
